src/main/java/io/camunda/zeebe/bpmnassert/testengine/IdleStateMonitor.java

```diff
@@ -7,13 +7,19 @@
 import io.camunda.zeebe.logstreams.storage.LogStorage;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
 
 final class IdleStateMonitor implements LogStorage.CommitListener, StreamProcessorListener {
 
+  private static final Timer TIMER = new Timer();
   private final List<Runnable> callbacks = new ArrayList<>();
   private final LogStreamReader reader;
-  private long lastEventPosition = -1L;
-  private long lastProcessedPosition = -1;
+  private volatile long lastEventPosition = -1L;
+  private volatile long lastProcessedPosition = -1L;
+
+  private volatile TimerTask task =
```

Comment:
How?

Reply:
This comment goes hand-in-hand with the comment about removing the synchronized. Then it'd end up like this:

```java
private void checkIdleState() {
  if (isInIdleState()) {
    final TimerTask task = createTask();
    try {
      TIMER.schedule(task, 10);
    } catch (IllegalStateException e) {
      // thrown - among others - if task was cancelled before it could be scheduled
      // do nothing in this case
    }
  }
}
```

This should be sufficient as the tasks should not have any impact on each other. I personally wouldn't mind risking it, and re-adding the `synchronized` if it turns out that it doesn't work as expected.

Reply:
Ah, now i see what you mean. 

No I think this won't work.

It might just so happen that when the task fires we have another temporary idle state. But without a way to cancel a running timer, we don't know whether the idle state really lasted all that time. 

I would not be comfortable with that change.

Reply:
That would be easily solved by re-adding the `else`:
```java
  private void checkIdleState() {
    final TimerTask task = createTask();
    if (isInIdleState()) {
      try {
        TIMER.schedule(task, 10);
      } catch (IllegalStateException e) {
        // thrown - among others - if task was cancelled before it could be scheduled
        // do nothing in this case
      }
    } else {
      task.cancel();
    }
  }
```

My main concern is with the `synchronized`. I find the code a bit confusing because of them. I also feel it might slow down the tests since the threads will be waiting to access the task. So the task will be locked each time we call `checkIdleState()` for 10ms.

Reply:
Nevermind the lock I mentioned. That only happens when there is an idle state, which should not occur too often.

Reply:
> That would be easily solved by re-adding the else:

:clown_face: 
whaaaaat? You want to cancel a task that you just created and that never got scheduled? I think you need to have lunch :-)

Reply:
> Nevermind the lock I mentioned. That only happens when there is an idle state, which should not occur too often.

'TIMER.schedule(task, 10)` doesn't block for 10 ms, it returns immediately

Reply:
🤦 Not my brightest moment